Keithley2612B_voltage_sweep.py

```diff
@@ -75,45 +72,119 @@ def get_sweep_type():
         else:
             print ('INVALID INPUT')
 
-def sweep_operation(smu_id, steps_no, measure_delay, nplc, start, end):
+def sweep_operation(smu_id, steps_no, measure_delay, nplc, start, end, scan_rate):
```

Comment:
Should I remove the parameter 'measure_delay' since it is not being used in the function definition?

Reply:
Yeah. Sorry I left it there, I was kinda rushing through the debugging. I replaced the measure_delay with source_delay, which is calculated in your sweep function based on the scan rate, so it is not needed anymore.

Reply:
No problem, I removed it.